flathunter/crawl_wggesucht.py

```diff
@@ -79,3 +81,29 @@ def load_address(self, url):
         address = ' '.join(response.find('div', {"class": "col-sm-4 mb10"})
                            .find("a", {"href": "#mapContainer"}).text.strip().split())
         return address
+
+    def get_soup_from_url(self, url, driver=None, captcha_api_key=None, checkbox=None, afterlogin_string=None):
+        """
+        Creates a Soup object from the HTML at the provided URL
+
+        Overwrites the method inherited from abstract_crawler. This is
+        necessary as we need to reload the page once for all filters to
+        be applied correctly on wg-gesucht.
+        """
+        self.rotate_user_agent()
+        sess = requests.session()
+        # First page load to set filters; response is discarded
+        sess.get(url, headers=self.HEADERS)
+        # Second page load
```

Comment:
It makes more sense to me to call 'super().get_soup_from_url' at this point to avoid the copy-paste from the superclass. Can you try with that?

Reply:
Thanks for your feedback! I totally agree that copying most of the method from the superclass is not very elegant. I tried with your suggestion like so:

```python
    def get_soup_from_url(self, url, driver=None, captcha_api_key=None, checkbox=None, afterlogin_string=None):
        """
        Creates a Soup object from the HTML at the provided URL

        Overwrites the method inherited from abstract_crawler. This is
        necessary as we need to reload the page once for all filters to
        be applied correctly on wg-gesucht.
        """
        self.rotate_user_agent()
        sess = requests.session()
        # First page load to set filters; response is discarded
        sess.get(url, headers=self.HEADERS)
        # Second page load
        return(super().get_soup_from_url(url))
```

It this what you meant? Unfortunately, this code also does not apply the filter correctly. I think the problem is that we need to persist the session (cookies I guess?) across both page loads for filtering to work properly. However, when we call the method from the superclass, the session doesn't carry over.

Otherwise, we might have to split `get_soup_from_url()` in `abstract_crawler.py` into two methods, one to get the response and another to evaluate it?

Reply:
Yeah. Okay. Not so easy. I guess it depends how deep you want to get into this. To just fix the bug, we can merge this and then we will have a maintenance headache in the future. To make this nice, I guess we need a more flexible / pluggable fetching architecture so that proxies, selenium support, sessions and retrying are all different fetch strategies that can be used (and maybe combined).

For now, I'll approve this - it works. If you or someone else is ambitious, you can look at building a more generic fetch architecture some other time. Thanks for having a go.